Jellyfin.Server/Migrations/MigrationRunner.cs

```diff
@@ -28,47 +29,42 @@ public static void Run(CoreAppHost host, ILoggerFactory loggerFactory)
             var logger = loggerFactory.CreateLogger<MigrationRunner>();
             var migrationOptions = ((IConfigurationManager)host.ServerConfigurationManager).GetConfiguration<MigrationOptions>(MigrationsListStore.StoreKey);
 
-            if (!host.ServerConfigurationManager.Configuration.IsStartupWizardCompleted && migrationOptions.Applied.Length == 0)
+            if (!host.ServerConfigurationManager.Configuration.IsStartupWizardCompleted && migrationOptions.Applied.Count == 0)
             {
                 // If startup wizard is not finished, this is a fresh install.
                 // Don't run any migrations, just mark all of them as applied.
                 logger.LogInformation("Marking all known migrations as applied because this is fresh install");
-                migrationOptions.Applied = Migrations.Select(m => m.Name).ToArray();
+                migrationOptions.Applied.AddRange(Migrations.Select(m => m.Id));
                 host.ServerConfigurationManager.SaveConfiguration(MigrationsListStore.StoreKey, migrationOptions);
                 return;
             }
 
-            var applied = migrationOptions.Applied.ToList();
-
             for (var i = 0; i < Migrations.Length; i++)
             {
-                var updater = Migrations[i];
-                if (applied.Contains(updater.Name))
+                var migrationRoutine = Migrations[i];
+                if (migrationOptions.Applied.Contains(migrationRoutine.Id))
                 {
-                    logger.LogDebug("Skipping migration '{Name}' since it is already applied", updater.Name);
+                    logger.LogDebug("Skipping migration '{Name}' since it is already applied", migrationRoutine.Name);
                     continue;
                 }
 
-                logger.LogInformation("Applying migration '{Name}'", updater.Name);
+                logger.LogInformation("Applying migration '{Name}'", migrationRoutine.Name);
+
                 try
                 {
-                    updater.Perform(host, logger);
+                    migrationRoutine.Perform(host, logger);
                 }
                 catch (Exception ex)
                 {
-                    logger.LogError(ex, "Could not apply migration '{Name}'", updater.Name);
+                    logger.LogError(ex, "Could not apply migration '{Name}'", migrationRoutine.Name);
                     throw;
                 }
 
-                logger.LogInformation("Migration '{Name}' applied successfully", updater.Name);
-                applied.Add(updater.Name);
-            }
-
-            if (applied.Count > migrationOptions.Applied.Length)
```

Comment:
Instead of waiting until all migrations are run to save the configuration, instead save it immediately after each migration is run. This is done in case a migration throws an exception, so we still know which ones executed successfully.